components/builder-protocol/src/originsrv.rs

```diff
@@ -260,19 +260,28 @@ impl Serialize for AccountOriginListResponse {
 }
 
 impl Routable for CheckOriginAccessRequest {
-    type H = u64;
+    type H = InstaId;
 
     fn route_key(&self) -> Option<Self::H> {
         // TODO:
-        Some(self.get_account_id())
+        Some(InstaId(self.get_origin_id()))
     }
 }
 
 impl Routable for OriginProjectGet {
     type H = String;
 
     fn route_key(&self) -> Option<Self::H> {
-        Some(String::from(self.get_name()))
+        let name = self.get_name();
+        let origin_name = match name.split('/').nth(0) {
+            Some(origin_name) => origin_name,
+            None => {
+                println!("Cannot route origin project get; malformed project name - routing on \
+                        screwedup to not kill the service");
+                "screwedup"
```

Comment:
The reality is that we need to change the message payload here - routing on a string that can be malformed is just a mistake, and we can easily force the correct `origin_id` into the message payload, letting us route on the `InstaId`, and removing this issue.